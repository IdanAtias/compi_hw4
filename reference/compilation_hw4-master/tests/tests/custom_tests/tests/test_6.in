// Matrix operations

matrix m_square(3,3) = [1, 2, 3; 4, 5, 6; 7, 8, 9];
matrix m(2,3) = [1, 2, 3; 4, 5, 6];

m_square[0, 0];
m_square[0, 1];
m_square[0, 2];
m_square[1, 0];
m_square[1, 1];
m_square[1, 2];
m_square[2, 0];
m_square[2, 1];
m_square[2, 2];

transpose(m);
transpose(m_square);

matrix I(3,3) = [1, 0, 0; 0, 1, 0; 0, 0, 1];

m_square + I;

m_square - I;

m_square :: I;
m_square :: transpose(m);

m_square * I;

m_square * m_square;
m * transpose(m);


-m;

int a = 3;

m * a;
a * m;
m * -1;
m * (-1);
m * 0;

m_square ^ (a);

