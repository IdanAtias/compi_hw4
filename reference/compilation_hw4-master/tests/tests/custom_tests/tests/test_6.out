m_square=
1,2,3
4,5,6
7,8,9
m=
1,2,3
4,5,6
1
2
3
4
5
6
7
8
9
1,4
2,5
3,6
1,4,7
2,5,8
3,6,9
I=
1,0,0
0,1,0
0,0,1
2,2,3
4,6,6
7,8,10
0,2,3
4,4,6
7,8,8
1,2,3,1,0,0
4,5,6,0,1,0
7,8,9,0,0,1
1,2,3,1,4
4,5,6,2,5
7,8,9,3,6
1,2,3
4,5,6
7,8,9
30,36,42
66,81,96
102,126,150
14,32
32,77
-1,-2,-3
-4,-5,-6
a=
3
3,6,9
12,15,18
3,6,9
12,15,18
-1,-2,-3
-4,-5,-6
-1,-2,-3
-4,-5,-6
0,0,0
0,0,0
468,576,684
1062,1305,1548
1656,2034,2412
